### PR TITLE
swift-format: new port

### PR DESCRIPTION
**Pull request by goranmoomin**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.4 20F71 arm64
Xcode 12.5 12E262

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/swift-format/Portfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+supported_archs     x86_64 arm64
+
+# Latest version gets no suffix
+github.setup        apple swift-format 0.50400.0
+set requiredXcodeVersion 12.5
+checksums           rmd160  beb5b6df42f722293097d7eed88ba320023b6bda \
+                    sha256  d8f128c1fb9a4c0f9d0bbebc4dbe6455dcb6026d7b279abbead4e1cb9d73b4de \
+                    size    227662
+
+subport ${name}54 {}
```
Comment (ryandesign) [inline lang/swift-format/Portfile:19]: Looks like the swift-format54 subport does the same thing as the main swift-format port? We don't want to have multiple ports that do the same thing. How about making the swift-format54 subport install that version, and make the main swift-format port be a stub that does nothing but install a readme file and declare a runtime dependency on swift-format54? We use this pattern elsewhere. See for example how the php port has subports and the main port just declares a dependency on php80.

Comment (goranmoomin) [reply lang/swift-format/Portfile:19]: @ryandesign This was my first portfile I've ever written – I'm reading the php port file, but it's too complex for me to grok how to do it.

Can you give me some pointers on how to do make the main port a stub and make it depend to a specific subport? Are there some portfiles that are less complex than the php port?

Comment (pmetzger) [reply lang/swift-format/Portfile:19]: We have a manual! The macports guide is actually pretty good! https://guide.macports.org/

Comment (goranmoomin) [reply lang/swift-format/Portfile:19]: I did read the Portfile reference part of the guide, but my experience was that it didn't have all the information that I needed?

E.g. I couldn't find much information on how subports work – the Portfile I currently have was written based on other Portfiles in the repository.

Comment (goranmoomin) [reply lang/swift-format/Portfile]: @ryandesign I've just updated the main port to be a stub one that just depends on swift-format54. I'm not sure if this is an idiomatic Portfile – could you please take a look at it?

**Pull request closed without merging**